api/v1alpha3/condition_consts.go

```diff
@@ -127,3 +124,21 @@ const (
 	// WaitingForRemediationReason is the reason used when a machine fails a health check and remediation is needed.
 	WaitingForRemediationReason = "WaitingForRemediation"
 )
+
+// Conditions and condition Reasons for the Machine's Node object
+const (
+	// MachineNodeHealthyCondition provides info about the operational state of the Kubernetes node hosted on the machine by summarizing  node conditions.
+	// If the conditions defined in a Kubernetes node (i.e., NodeReady, NodeMemoryPressure, NodeDiskPressure, NodePIDPressure, and NodeNetworkUnavailable) are in a healthy state, it will be set to True.
+	MachineNodeHealthyCondition ConditionType = "NodeHealthy"
```

Comment:
Should we? Doesn't NodeReady require CNI?

Reply:
Yes, this is debatable, however, I'm inclined to have this included because:
- installing CNI could be easily achieved during cluster provisioning via ClusterResourceSet, so I'm expecting Node ready should happen immediately after infrastructure provisioning
- in the conditions proposal we are stating 
  - A Ready condition SHOULD be provided at object level to represent the overall operational state of the component (and IMO for a machine, this includes having the node with a CNI)
  - The Ready condition MUST be based on the summary of more detailed conditions existing on the same object
  - An object SHOULD NEVER be in status Ready=True if one of the object's conditions are false or if one of the object dependents is in status Ready=False.

However, I'm testing this, so I can have real feedback on it

Reply:
This is a contract change and should be discussed in the community, let's open an issue please?

Reply:
From my testing, this provides a nice UX for the users, with the only minus that we are not watching nodes in the remote cluster and so sometimes it takes a little to get synchronized.
Happy to demo it if it can help

Reply:
Happy to have a demo, I'd still like for us to keep the discussion and proposal in a new issue outlining the contract changes for v1alpha4

Reply:
Let's tackle this in a separated PR